<launch>
  <include file="$(find spirit_utils)/launch/load_params.launch" />
  <param name="robot_description_sdf" textfile="$(find spirit_description)/sdf_mesh/spirit.sdf" />

  <!-- Spawn SDF model -->
  <node name="spawn_sdf_model" pkg="gazebo_ros" type="spawn_model" args="-sdf -param robot_description_sdf -model spirit -x 0.0 -y 0.0 -z 0.4" output="screen"/>

  <rosparam file="$(find gazebo_scripts)/config/spirit_control.yaml" command="load"/>

  <node name="controller_spawner"
        pkg="controller_manager"
        type="spawner"
        respawn="false"
        output="screen"
        ns="/spirit"
        args="joint_controller joint_state_controller">
  </node>

  <!-- Load the contact state publisher -->
  <node name="contact_state_publisher" pkg="gazebo_scripts" type="contact_state_publisher_node" output="screen"/>
 
</launch>